com.civicrm.hrjobroles/tests/phpunit/api/v3/HrJobRolesTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+require_once 'HrJobRolesTestBase.php';
+
+class api_v3_HrJobRolesTest extends HrJobRolesTestBase {
+
+
+  function setUp() {
+    parent::setUp();
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  /**
+   * check with empty array
+   */
+  function testJobRoleCreateEmpty() {
+    $this->callAPIFailure('HrJobRoles', 'create', array());
+  }
+
+  /**
+   * check if required fields are not passed
+   */
+  function testJobRoleCreateWithoutJobContract() {
+    $params = array(
+      'title' => 'test role',
+    );
+    $this->callAPIFailure('HrJobRoles', 'create', $params);
+  }
+
+  /**
+   *  Test creating new role with valid parameters
+   */
+  function testJobRoleCreate() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    $params = ['job_contract_id' => $contract->id, 'title' => 'test role', 'sequential' => 1];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['job_contract_id'], $contract->id);
+    $this->assertEquals($result['values'][0]['title'], 'test role');
+  }
+
+  /**
+   *  Test creating new role with valid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithValidOptionValues() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // create option group and values
+    $this->createSampleOptionGroupsAndValues();
+
+    $params = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test role',
+      'location' => 'amman',
+      'region' => 'south amman',
+      'department' => 'amman devs',
+      'level_type' => 'guru',
+      'cost_center' => 'abdali',
+      'sequential' => 1
+    ];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['location'], 'amman');
+    $this->assertEquals($result['values'][0]['region'], 'south amman');
+    $this->assertEquals($result['values'][0]['department'], 'amman devs');
+    $this->assertEquals($result['values'][0]['level_type'], 'guru');
+    $this->assertEquals($result['values'][0]['cost_center'], 'abdali');
+  }
+
+  /**
+   *  Test creating new role with Invalid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithInvalidOptionValues() {
+    // create contact
```

Comment:
So what you're actually doing here is testing code from the core, not ours. One more reason to remove these tests, since this should already be covered by the tests in core.